### PR TITLE
:bug: fix(#22) missing package.json or node_modules

### PR DESCRIPTION
**Pull request by bpetetot**
#22

### CODE REVIEW

**Comment by bpetetot** [comment]
@fabienjuif : I have added all tests for `modules.js` with 100% coverage :)
Can you tell me what do think about it please ? (about technical implementation and use cases)

File: src/server/webpack/modules.spec.js [hunk 1]
```diff
@@ -1,8 +1,110 @@
...
+    const paths = await service.get(component)
+
+    // asserts
+    expect(loglevel.warn.mock.calls.length).toBe(1)
+    expect(loglevel.warn.mock.calls[0][0]).toBe('package.json not found, unable to identify project root')
+    expect(paths).toBeDefined()
+    expect(paths.length).toBe(0)
+  })
+
+  it('should show warning when node_modules not found', async () => {
```
Comment (fabienjuif) [inline src/server/webpack/modules.spec.js]: I would add a `describe('errors', () => {` to handle errors.

File: src/server/webpack/modules.spec.js [hunk 2]
```diff
@@ -1,8 +1,110 @@
...
+  })
+
+  it('should throw an error when unexpected fs error happens', async () => {
+    // mocks
+    fs.readdirAsync.mockImplementation(jest.fn(async () => ['package.json']))
+    fs.statAsync.mockImplementation(jest.fn(async () => {
+      throw new Error()
+    }))
+
+    // calls
```
Comment (fabienjuif) [inline src/server/webpack/modules.spec.js]: If there is no error your test will pass.
That's why I use a boolean to test this kind of code : 
```es6
error = false
try {
// ...
} catch (ex) {
error = true
// ...
}
expect(error).toBe(true)
```

**Review by fabienjuif — CHANGES_REQUESTED** [review]
Great :)

I think we don't have a test that iterate more than once on the path finding.

**Comment by bpetetot** [comment]
@fabienjuif : tests updated, you can do a final check

**Comment by bpetetot** [comment]
@fabienjuif : done

File: src/server/webpack/modules.spec.js [hunk 3]
```diff
@@ -1,8 +1,131 @@
...
+    const paths = await service.get(component)
+
+    // asserts
+    expect(paths).toBeDefined()
+    expect(paths).toContain('/foo/bar')
+    expect(paths).toContain('/foo/bar/node_modules')
+  })
+
+  it('should iterate over the path until it finds a package.json file', async () => {
```
Comment (fabienjuif) [inline src/server/webpack/modules.spec.js:39]: 👍

**Review by fabienjuif — APPROVED** [review]
Waiting for coverage

**Pull request merged**